assets/js/script.js

```diff
@@ -14,3 +72,75 @@ closeX.addEventListener('click', toggleMenu);
 menuItems.forEach((e) => {
   e.addEventListener('click', toggleMenu);
 });
+
+for (let i = 0; i < projects.length; i += 1) {
+  const cardDiv = document.createElement('div');
+  cardDiv.className = `card card${i + 1}`;
+  let li = '';
+  const eachObj = projects[i];
+  const tech = projects[i].technologies;
+  for (let j = 0; j < tech.length; j += 1) {
+    li += `
+<li class="card-list-item">${tech[j]}</li>
+`;
+  }
+  cardDiv.innerHTML = `
+          <div class="card-image image${i + 1}">
+              <img src="${eachObj.featured_image}" alt="">
+            </div>
+            <div class="card-body">
+              <h3 class="project-title">${eachObj.title}</h3>
+              <ul class="card-list">
+              ${li}
+              </ul>
+              <button myIndex=${i} type="button" class="card-btn">
+                See this project &rarr;
+              </button>
+          </div>
+`;
+  cardContainer.append(cardDiv);
+}
+
+const modal = document.querySelector('.modal');
+
+const showModal = () => {
+  const btn = document.querySelectorAll('.card-btn');
+  btn.forEach((e) => {
+    e.addEventListener('click', () => {
+      const index = e.getAttribute('myIndex');
+      modal.classList.toggle('hidden');
+      const tech = projects[index].technologies;
+      let li = '';
+      for (let i = 0; i < tech.length; i += 1) {
+        li += `
+<li class="modal-list-item">${tech[i]}</li>
+`;
+      }
+      modal.innerHTML = `
+      <div class="modal-content">
+      <div class="modal-header">
+        <span class="closeModal"><i class="fa-solid fa-xmark fa-2xl"></i></span>
+        <h2>${projects[index].title}</h2>
+        <ul class="modal-list">
+          ${li}
+        </ul>
+      </div>
+      <div class="modal-body">
+        <img src="${projects[index].featured_image}" alt="Project image">
+        <p>${projects[index].description}</p>
+        <div class="button-wrapper">
+          <button type="button" class="modal-button">See live<i class="fa-solid fa-arrow-up-right-from-square"
+              data-fa-transform="grow-15 right-100"></i></button><button type="button" class="modal-button">See source<i
+              class="modal-icon fa-brands fa-github" data-fa-transform="grow-15 right-90"></i></button>
```

Comment:
- [ ] Kudos on properly implementing JS code. Everything works as expected 👏🏻 . The only thing remaining is adding hover effects on buttons as provided on the design :
<img width="232" alt="Screenshot 2022-09-07 at 22 13 27" src="https://user-images.githubusercontent.com/77038610/188958781-13394535-02db-4bfd-a454-bfd8589acaea.png">